src/Dropdown.svelte

```diff
@@ -0,0 +1,68 @@
+<script>
+	import clsx from 'clsx';
+
+	import { context } from './DropdownContext';
```

Comment:
Hi @TheBosZ , after looking at this (and catching up with Svelte 3), it looks like the issue is the Dropdown state is shared here.  It looks like all instances are importing the same store instance?

If so can we change this to not use a shared store?  E.g. a local to Dropdown `let isOpen`, etc.
I attempting to change but having trouble with the context here, context.update, etc

Reply:
The point of the Dropdown context is to wrap up the shared stuff between all the dropdown components. Reactstrap uses React context to do it.

My knowledge of Svelte isn't super great so I just did a naive implementation.

Now that I've thought about it for a week or so, it would be better to implement it using events.

If I get a free bit of time, I'll see if I can re-implement it.

Reply:
I jumped the gun on how to solve this.

Let me start by explaining what Reacstrap does.

In Reactstrap, the `<Dropdown>` component is really just a container component for `<DropdownToggle>` and `<DropdownMenu>`. It doesn't do too much other than set up the two child components and add event listeners.

The dropdown components share state via a `Context`. The context hold 4 fields: `toggle`, `isOpen`, `direction`, and `inNavbar`. We can ignore `direction` and `inNavbar` for now. They're really only there to help the components figure out what to tell PopperJS.

`toggle` is used to open and close the menu from `<DropdownToggle>`. `isOpen` is used to tell the menu itself to open or close. (Essentially `toggle` will flip `isOpen` between true and false).

Onto my Sveltestrap implementation.

I naively tried to re-implement the Context idea using a [Svelte store](https://svelte.dev/docs#svelte_store). Unfortunately, as you've experienced, this state is **global** and not local to the component and its children.

I can **not** figure out how to make a store that is local to a component and its children. If I were to ask, I might be directed to [this issue](https://github.com/sveltejs/svelte/issues/1435) which doesn't actually answer the problem.

For a solution:
We need to at least replicate the two properties `toggle` and `isOpen`.

I originally thought I could replicate the `toggle` property using Component Events. Turns out that this doesn't work because you can't attach listeners to slots.

For the `isOpen` prop, I thought we could at least propagate the `isOpen` prop from `<Dropdown>` but it turns out that props on slots mean [something different](https://svelte.dev/docs#Slots).

I'm at a loss on how to fix this. My next step would be to get into their Discord and ask someone on the team how to do this.

Unfortunately, my team has decided to not move forward with integrating Svelte into our React app so I'm not going to be able to continue to work on this.

Reply:
Hi @TheBosZ, thanks for the info and the update.  No worries, I will continue looking at a solution, thank you for the PR and we'll merge after resolving this above.